### PR TITLE
neatvi: new package at 07

### PR DESCRIPTION
**Pull request by aaronNGi**
## Description of package

A small vi/ex editor for editing bidirectional UTF-8 text http://litcave.rudi.ir/

## Checklist

- [x] Latest upstream version.
- [x] I agree to maintain this package (**required**).
- [x] I agree to notify KISS if I can no longer maintain this package (**required**).

### CODE REVIEW

File: community/neatvi/build [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh -e
+
+sed -i -e '/CFLAGS = /c\' -e 'CFLAGS += -Wall' Makefile
```
Comment (dylanaraps) [inline community/neatvi/build]: Any reason for this line?

File: community/neatvi/build [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+sed -i s/0600/0666/ ex.c
```
Comment (dylanaraps) [inline community/neatvi/build]: Same goes for this one.

File: community/neatvi/build [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+make
+install -Dm 755 -- vi "$1"/usr/bin/neatvi
```
Comment (dylanaraps) [inline community/neatvi/build]: `--` can be removed.

File: community/neatvi/build (hunk 1 above)
Comment (aaronNGi) [reply community/neatvi/build]: It's overwriting the global CFLAGS otherwise, no?

File: community/neatvi/build (hunk 2 above)
Comment (aaronNGi) [reply community/neatvi/build]: `neatvi` creates files with mode 0600 by default, I don't like it being so strict. Do you want comments for things like this?

File: community/neatvi/build (hunk 1 above)
Comment (dylanaraps) [reply community/neatvi/build]: You are correct, though it's also adding `-Wall`?

Could this not be: `sed -i 's/CFLAGS =/CFLAGS +=/' Makefile`?

File: community/neatvi/build (hunk 2 above)
Comment (dylanaraps) [reply community/neatvi/build]: Omitting comments is fine so long as there's a written explanation *somewhere* (Comments on this PR for example). This addition may also be temporary if the developer changes it to `0666`.  :+1:

File: community/neatvi/build (hunk 1 above)
Comment (aaronNGi) [reply community/neatvi/build]: No, the original Makefile also sets `-O2`. Appending that to `CFLAGS` overwrites the users optimization level. My first idea was to simply remove that entire line but thought maybe it's better to retain warnings. Two substitutions, one to change `=` to `+=`, and one to remove `-O2` would have worked as well but I found this prettier.

File: community/neatvi/build (hunk 2 above)
Comment (aaronNGi) [reply community/neatvi/build]: Got it.

Also asked the author to make that change.

**Comment by dylanaraps** [comment]
Thanks :+1:

**Pull request merged**